File: app/src/main/java/name/monwf/customiuizer/prefs/SpinnerExFake.java
package name.monwf.customiuizer.prefs;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.util.AttributeSet;
import android.util.Pair;

import java.util.ArrayList;

import name.monwf.customiuizer.utils.AppHelper;

public class SpinnerExFake extends SpinnerEx {

	private String value = null;
	private final ArrayList<Pair<String, String>> others = new ArrayList<Pair<String, String>>();

	public SpinnerExFake(Context context, AttributeSet attrs) {
		super(context, attrs);
		others.clear();
	}

	public void setValue(String val) {
		value = val;
	}

	public String getValue() {
		return value;
	}

	public void addValue(String key, String val) {
		if (val == null) val = AppHelper.getStringOfAppPrefs(key, null);
		if (val != null) others.add(new Pair<String, String>(key, val));
	}

	public void addValue(String key, Intent val) {
		String sVal;
		if (val == null)
			sVal = AppHelper.getStringOfAppPrefs(key, null);
		else
			sVal = val.toUri(0);
		if (sVal != null) others.add(new Pair<String, String>(key, sVal));
	}

	public void applyOthers() {
		if (others.size() == 0) return;
		SharedPreferences.Editor editor = AppHelper.appPrefs.edit();
		for (Pair<String, String> pref: others) {
			editor.putString(pref.first, pref.second);
		}
		editor.apply();
	}

}

File: app/src/main/java/tv/withaibuild/customiuizer/prefs/SpinnerExFake.kt
package tv.withaibuild.customiuizer.prefs

import android.content.Context
import android.content.Intent
import android.util.AttributeSet
import android.util.Pair
import tv.withaibuild.customiuizer.utils.AppHelper

class SpinnerExFake(context: Context, attrs: AttributeSet?) : SpinnerEx(context, attrs) {

    var value: String? = null
    private val others = ArrayList<Pair<String, String>>()

    fun addValue(key: String, newValue: String?) {
        val v = newValue ?: AppHelper.getStringOfAppPrefs(key, null)
        if (v != null) others.add(Pair(key, v))
    }

    fun addValue(key: String, newValue: Intent?) {
        val sVal = if (newValue == null) {
            AppHelper.getStringOfAppPrefs(key, null)
        } else {
            newValue.toUri(0)
        }
        if (sVal != null) others.add(Pair(key, sVal))
    }

    fun applyOthers() {
        if (others.isEmpty()) return
        val editor = AppHelper.appPrefs.edit()
        for (pref in others) {
            editor.putString(pref.first, pref.second)
        }
        editor.apply()
    }
}
